src/main/java/net/fabricmc/base/loader/ModInfo.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.base.loader;
+
+import com.github.zafarkhaja.semver.Version;
+import com.google.gson.*;
+import net.fabricmc.base.Side;
+
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ModInfo {
+
+//	Required
+	private String id;
+	private String group;
+	private Version version;
+
+//	Optional
+	private String modClass = "";
```

Comment:
This is ugly. We should process jars for classes with `@Mod(id = "examplemod")` annotations.

The id **can** then be matched up to a mod in the `mod.json`, however all the stuff should be settable through the annotation. With values in the `mod.json` overriding the ones in the annotation.